drivers/pwm/pwm_mcux.c

```diff
@@ -35,34 +36,17 @@ struct pwm_mcux_config {
 struct pwm_mcux_data {
 	uint32_t period_cycles[CHANNEL_COUNT];
 	pwm_signal_param_t channel[CHANNEL_COUNT];
+	struct k_mutex lock;
```

Comment:
Have you considered using the more lightweight `struct k_sem` for locking?

Reply:
I just looked into it. I guess it would give up checking if the same thread gave and released the lock, but if you think it would be a benefit for speed, space due to keeping the thread pointer around, happy to make the change.

Reply:
I also think we want priority inheritance here from the mutex. We discovered the need for a lock  when running a buzzer and a motor driver both on pwm with the pwm_mcux driver.  I don't want a motor write to be waiting around for a buzzer write and get in a priority inversions situation.

Reply:
the mutex is fine to me, but just curious, why did a thread controlling a buzzer have a higher priority than the motor controlling thread?

Reply:
The motor thread had a higher priority. We didn't have a mutex or sem before. I think there must have been a context switch to the motor write during the buzzer write or something similar that was causing issues. I had the threads set to prio 1 for motors, 4 for buzzers I think. So they were set to be cooperative.